plugins/mac-capture/mac-screen-capture.m

```diff
@@ -0,0 +1,1143 @@
+				  obs_module_text("WindowCapture"), 1);
+	obs_property_list_add_int(capture_type,
+				  obs_module_text("ApplicationCapture"), 2);
+
+	obs_property_set_modified_callback2(capture_type, capture_type_changed,
+					    data);
+
+	obs_property_t *display_list = obs_properties_add_list(
+		props, "display", obs_module_text("DisplayCapture.Display"),
+		OBS_COMBO_TYPE_LIST, OBS_COMBO_FORMAT_INT);
+
+	obs_property_t *app_list = obs_properties_add_list(
+		props, "application", obs_module_text("Application"),
+		OBS_COMBO_TYPE_LIST, OBS_COMBO_FORMAT_STRING);
+
+	obs_property_t *window_list = obs_properties_add_list(
+		props, "window", obs_module_text("WindowUtils.Window"),
+		OBS_COMBO_TYPE_LIST, OBS_COMBO_FORMAT_INT);
+
+	obs_property_t *empty = obs_properties_add_bool(
+		props, "show_empty_names",
+		obs_module_text("WindowUtils.ShowEmptyNames"));
+
+	obs_property_t *hidden = obs_properties_add_bool(
+		props, "show_hidden_windows",
+		obs_module_text("WindowUtils.ShowHidden"));
+
+	obs_property_set_modified_callback2(hidden, content_settings_changed,
+					    sc);
+
+	obs_properties_add_bool(props, "show_cursor",
+				obs_module_text("DisplayCapture.ShowCursor"));
+
+	switch (sc->capture_type) {
+	case 0: {
+		obs_property_set_visible(display_list, true);
+		obs_property_set_visible(window_list, false);
+		obs_property_set_visible(app_list, false);
+		obs_property_set_visible(empty, false);
+		obs_property_set_visible(hidden, false);
+		break;
+	}
+	case 1: {
+		obs_property_set_visible(display_list, false);
+		obs_property_set_visible(window_list, true);
+		obs_property_set_visible(app_list, false);
+		obs_property_set_visible(empty, true);
+		obs_property_set_visible(hidden, true);
+		break;
+	}
+	case 2: {
+		obs_property_set_visible(display_list, true);
+		obs_property_set_visible(app_list, true);
+		obs_property_set_visible(window_list, false);
+		obs_property_set_visible(empty, false);
+		obs_property_set_visible(hidden, false);
+		break;
+	}
+	}
+
+	obs_property_set_modified_callback2(empty, content_settings_changed,
+					    sc);
+
+	content_changed(sc, props);
+
+	if (@available(macOS 13.0, *))
+		;
+	else
+		obs_properties_add_text(props, "audio_info",
+					obs_module_text("SCKAudioUnavailable"),
+					OBS_TEXT_INFO);
+
+	return props;
+}
+
+struct obs_source_info screen_capture_info = {
+	.id = "screen_capture",
+	.type = OBS_SOURCE_TYPE_INPUT,
+	.get_name = screen_capture_getname,
+
+	.create = screen_capture_create,
+	.destroy = screen_capture_destroy,
+
+	.output_flags = OBS_SOURCE_VIDEO | OBS_SOURCE_CUSTOM_DRAW |
+			OBS_SOURCE_DO_NOT_DUPLICATE | OBS_SOURCE_SRGB |
+			OBS_SOURCE_AUDIO,
+	.video_tick = screen_capture_video_tick,
+	.video_render = screen_capture_video_render,
+
+	.get_width = screen_capture_getwidth,
+	.get_height = screen_capture_getheight,
+
+	.get_defaults = screen_capture_defaults,
+	.get_properties = screen_capture_properties,
+	.update = screen_capture_update,
+	.icon_type = OBS_ICON_TYPE_GAME_CAPTURE,
```

Comment:
I find the usage of the game capture icon a bit weird, feels like it should be window capture or maybe display capture instead - or was there some other motivation behind using the game capture icon?

Reply:
My hunch is that SCK was mainly used for capturing games at Apple because of OBS historic main usage as a game capture tool. As it's combining game capture, desktop capture, window capture, and a never before available (not even on Windows) app capture, I dunno what the right icon would be.

Reply:
FWIW Syphon source also has the game capture icon, so I think it works within the icon constraints that we have.

Reply:
Syphon having the game capture icon is a relict of its past as a game capture source in combination with syphon inject, and I think it was discussed before that the icon isn’t really optimal there anymore.

As for here, my feeling is that either window or display capture would be better (I’d personally probably go for window capture): If someone does indeed use it for game capturing, looking at the sources list they will likely understand that this can capture their screen and as such their game. On the opposite way, not so much. I imagine someone just looking to record their Safari will be confused if they have to use the source with the game controller icon.

Reply:
I've opted for the desktop capture icon for now, because it's based on ScreenCaptureKit and includes desktop capture.

Reply:
Sounds good, either should be fine